modules/atlaslocal/atlaslocal.go

```diff
@@ -0,0 +1,360 @@
+			return fmt.Errorf("password file does not exist or is not accessible: %w", err)
+		}
+
+		if info.IsDir() {
+			return fmt.Errorf("password file must be a file, got a directory: %s", passwordFile)
+		}
+
+		req.Env["MONGODB_INITDB_ROOT_PASSWORD_FILE"] = passwordFile
+
+		// Mount password file as read-only in the container.
+		prev := req.HostConfigModifier
+		req.HostConfigModifier = func(hostConfig *container.HostConfig) {
+			if prev != nil {
+				prev(hostConfig)
+			}
+
+			if passwordFile != "" {
+				hostConfig.Binds = append(hostConfig.Binds, fmt.Sprintf("%s:%s:ro", passwordFile, passwordFile))
+			}
+		}
+
+		return nil
+	}
+}
+
+// WithDisableTelemetry opts out of telemetry for the MongoDB Atlas Local
+// container by setting the DO_NOT_TRACK environment variable to 1.
+func WithDisableTelemetry() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.Env["DO_NOT_TRACK"] = "1"
+
+		return nil
+	}
+}
+
+// WithInitDatabase sets MONGODB_INITDB_DATABASE environment variable so the
+// init scripts and the default connection string target the specified database
+// instead of the default "test" database.
+func WithInitDatabase(database string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		req.Env["MONGODB_INITDB_DATABASE"] = database
+
+		return nil
+	}
+}
+
+// WithInitScripts mounts a directory containing .sh/.js init scripts into
+// /docker-entrypoint-initdb.d so they run in alphabetical order on startup. If
+// called multiple times, this funcion removes any prior init-scripts bind and
+// uses only the latest on specified.
+func WithInitScripts(scriptsDir string) testcontainers.CustomizeRequestOption {
+	abs, err := filepath.Abs(scriptsDir)
+	if err != nil {
+		return func(req *testcontainers.GenericContainerRequest) error {
+			return fmt.Errorf("get absolute path of scripts directory: %w", err)
+		}
+	}
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		prev := req.HostConfigModifier
+		req.HostConfigModifier = func(hostConfig *container.HostConfig) {
+			if prev != nil {
+				prev(hostConfig)
+			}
+
+			// Remove any old /docker-entrypoint-initdb.d bind mounts.
+			filtered := hostConfig.Binds[:0]
+			for _, bind := range hostConfig.Binds {
+				if !strings.HasSuffix(bind, ":/docker-entrypoint-initdb.d:ro") {
+					filtered = append(filtered, bind)
+				}
+			}
+			hostConfig.Binds = filtered
+
+			// Mount the new scriptsDir
+			hostConfig.Binds = append(hostConfig.Binds,
+				fmt.Sprintf("%s:/docker-entrypoint-initdb.d:ro", abs))
+		}
+
+		return nil
+	}
+}
+
+// WithMongotLogFile sets the path to the file where you want to store the logs
+// of Atlas Search (mongot) by setting the MONGOT_LOG_FILE environment variable.
+// Note that this can be set to /dev/stdout or /dev/stderr for convenience.
+func WithMongotLogFile(logFile string) testcontainers.CustomizeRequestOption {
```

Comment:
question: should we simplify the API adding types for stdout/stderr so users don't configure a non existing path? same for the runner log file